pycqed/instrument_drivers/meta_instrument/heterodyne.py

```diff
@@ -111,12 +112,21 @@ def common_init(self, name, LO, AWG, acquisition_instr='CBox',
         self._frequency = 5e9
         self.frequency(5e9)
         self.f_RO_mod(10e6)
-        self._eps = 0.01 # Hz slack for comparing frequencies
-        self._acq_marker_channels = 'ch4_marker1,ch4_marker2,' \
-                                    'ch3_marker1,ch3_marker2'
-
+        self._eps = 0.01  # Hz slack for comparing frequencies
+        self._acq_marker_channels = ('ch4_marker1,ch4_marker2,' +
+                                     'ch3_marker1,ch3_marker2')
 
     def prepare(self, get_t_base=True):
+        # Uploading the AWG sequence
+        if (self._awg_seq_filename not in self.AWG.setup_filename() or
+                self._awg_seq_parameters_changed) and self.auto_seq_loading():
+            self._awg_seq_filename = \
```

Comment:
This change breaks `LO_modulated_Heterodyne.prepare` function that currently inherits the base `prepare` function from  `Heterodyne` by programming the Tektronix with a wrong sequence. I suggest either overriding `LO_modulated_Heterodyne.prepare` or moving the Tektronix AWG programming back to instrument specific prepare functions.